docs: align landing pages with model (#2272)
canonical/charmcraft: docs/reference/index.rst

> Reference

Author: I don't love how granular I got with these sections. Curious to hear if anyone has recommendations for broader sections.

Reviewer: The only thing I'd advise against is using toctrees for these lists. It spreads the functional surface area and ties the ordering in the TOC to the ordering in the landing page, which isn't always going to be the same, especially in cases where there are pages we don't want to mention in the landing.

I'd consider putting the toctree at the end like usual and maintaining these links as independent lists.

Author: Good to know. I'd seen it done both ways and wasn't sure which was preferred. I'll make that change shortly.

=========

The following reference material describes all of Charmcraft's key components and

> processes. This includes commands, extensions, charm project files, and more.

Reviewer: ```suggestion
processes. It includes commands, extensions, charm project files, and more.
```

Author: https://github.com/canonical/charmcraft/pull/2272/commits/a200349c7334c7deaa19accd24d4e959a17cf41a

Command-line reference
----------------------

The following pages document all of Charmcraft's commands, including their usage,
options, and arguments:

.. toctree::
    :maxdepth: 2

    commands/index

Files
-----

In the context of Charmcraft, a *file* refers to any file in a project that can be

> initialised or packed. Such files have their contents documented in the following

Reviewer: ```suggestion
initialized or packed. The files are documented in the following
```

Author: https://github.com/canonical/charmcraft/pull/2272/commits/a200349c7334c7deaa19accd24d4e959a17cf41a

pages:

.. toctree::
    :maxdepth: 2

    files/index

Plugins and extensions
----------------------

> Charmcraft's extensions allow you to initialize your project with template YAML and

Reviewer: ```suggestion
Extensions help initialize your project with template YAML and
```

Author: https://github.com/canonical/charmcraft/pull/2272/commits/a200349c7334c7deaa19accd24d4e959a17cf41a

Ops-powered Python to remove the boilerplate steps of crafting charms for Django,
FastAPI, Flask, and Go applications.

.. toctree::
    :maxdepth: 2

    profile
    extensions/index
    plugins/index

Platforms
---------

A charm project file's ``platforms`` key defines where the charm will be built and
where it can run.

.. toctree::
    :maxdepth: 2

    platforms

Parts
-----

Parts provide a mechanism for your charm to obtain and process data from various
sources.

.. toctree::
    :maxdepth: 2

    parts/index

Changelog
---------

All changes to Charmcraft prior to version 3.4 are documented in the page below. For any
newer changes, see :ref:`release-notes`.

.. toctree::
    :maxdepth: 1

>     changelog

Reviewer: I wouldn't include these, as they already have their own landing page. Different category.

Author: https://github.com/canonical/charmcraft/pull/2272/commits/a200349c7334c7deaa19accd24d4e959a17cf41a

.. toctree::
    :hidden:

    analyzers-and-linters

>     models/index

Author: I couldn't think of anywhere meaningful to put these. Open to suggestions.

Reviewer: As suggested in my earlier comment, this is fine and expected. We can't include everything in the landing page.
